tests/Unit/Elliptic/BoundaryConditions/Test_AnalyticSolution.cpp

```diff
@@ -0,0 +1,145 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <algorithm>
+#include <cstddef>
+#include <string>
+#include <unordered_map>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/FaceNormal.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Tags.hpp"
+#include "Elliptic/BoundaryConditions/AnalyticSolution.hpp"
+#include "Elliptic/BoundaryConditions/ApplyBoundaryCondition.hpp"
+#include "Elliptic/BoundaryConditions/BoundaryCondition.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "Utilities/Literals.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace elliptic::BoundaryConditions {
+
+namespace {
+
+template <size_t N>
+struct ScalarFieldTag : db::SimpleTag {
+  static std::string name() noexcept { return "Field" + std::to_string(N); }
+  using type = Scalar<DataVector>;
+};
+
+template <size_t N>
+struct FluxTag : db::SimpleTag {
+  using type = tnsr::I<DataVector, 1>;
+};
+
+struct System {
+  static constexpr size_t volume_dim = 1;
+  using primal_fields = tmpl::list<ScalarFieldTag<1>, ScalarFieldTag<2>>;
+  using primal_fluxes = tmpl::list<FluxTag<1>, FluxTag<2>>;
+};
+
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Elliptic.BoundaryConditions.AnalyticSolution",
+                  "[Unit][Elliptic]") {
+  // Test factory-creation
+  using Registrar = Registrars::AnalyticSolution<System>;
+  const auto boundary_condition_base = TestHelpers::test_factory_creation<
+      BoundaryCondition<1, tmpl::list<Registrar>>>(
+      "AnalyticSolution:\n"
+      "  Field1: Dirichlet\n"
+      "  Field2: Neumann");
+
+  // Test semantics
+  using BoundaryConditionType =
+      typename Registrar::template f<tmpl::list<Registrar>>;
+  REQUIRE(dynamic_cast<const BoundaryConditionType*>(
+              boundary_condition_base.get()) != nullptr);
+  const auto& boundary_condition =
+      dynamic_cast<const BoundaryConditionType&>(*boundary_condition_base);
+  test_serialization(boundary_condition);
+  test_copy_semantics(boundary_condition);
+  auto move_boundary_condition = boundary_condition;
+  test_move_semantics(std::move(move_boundary_condition), boundary_condition);
+
+  // Test applying the boundary conditions
+  const Mesh<1> mesh{3, Spectral::Basis::Legendre,
```

Comment:
please add a test for at least 2d, ideally 3d too. 1d is very special/simple since there's only a point on the boundary.